packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts

```diff
@@ -0,0 +1,197 @@
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: undefined,
+      };
+
+      expect(elementHasNoQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has no query property', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+      };
+
+      expect(elementHasNoQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has empty string query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: '',
+      };
+
+      expect(elementHasNoQuery(element)).toBe(true);
+    });
+
+    it('should return false when element has non-empty query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: 'search query',
+      };
+
+      expect(elementHasNoQuery(element)).toBe(false);
+    });
+
+    it('should return false when element has whitespace-only query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: '   ',
+      };
+
+      expect(elementHasNoQuery(element)).toBe(false);
+    });
+  });
+
+  describe('#elementHasQuery', () => {
+    it('should return false when element has undefined query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: undefined,
+      };
+
+      expect(elementHasQuery(element)).toBe(false);
+    });
+
+    it('should return false when element has no query property', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+      };
+
+      expect(elementHasQuery(element)).toBe(false);
+    });
+
+    it('should return false when element has empty string query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: '',
+      };
+
+      expect(elementHasQuery(element)).toBe(false);
+    });
+
+    it('should return true when element has non-empty query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: 'search query',
+      };
+
+      expect(elementHasQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has whitespace-only query', () => {
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: '   ',
+      };
+
+      expect(elementHasQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has numeric zero as query', () => {
```

Comment:
We don't have a test equivalent to this one for the elementHasNoQuery method... but at this rate we'd also want to check the 'null' and 'undefined' strings 😅 

TBH I think this test is overkill and we should just remove it.